openedx/core/djangoapps/content/search/plain_text_math.py

```diff
@@ -0,0 +1,105 @@
+"""
+Helper class to convert mathjax equations to plain text.
+"""
+
+import re
+
+import unicodeit
+
+
+class PlainTextMath:
+    """
+    Converts mathjax equations to plain text using unicodeit and some preprocessing.
+    """
+    equation_pattern = re.compile(
+        r'\[mathjaxinline\](.*?)\[\/mathjaxinline\]|\[mathjax\](.*?)\[\/mathjax\]|\\\((.*?)\\\)|\\\[(.*?)\\\]'
+    )
+    eqn_replacements = (
+        # just remove prefix `\`
+        ("\\sin", "sin"),
+        ("\\cos", "cos"),
+        ("\\tan", "tan"),
+        ("\\arcsin", "arcsin"),
+        ("\\arccos", "arccos"),
+        ("\\arctan", "arctan"),
+        ("\\cot", "cot"),
+        ("\\sec", "sec"),
+        ("\\csc", "csc"),
+        # Is used for matching brackets in mathjax, should not be required in plain text.
+        ("\\left", ""),
+        ("\\right", ""),
+    )
+    regex_replacements = (
+        # Makes text bold, so not required in plain text.
+        (re.compile(r'\\mathbf{(.*?)}'), r"\1"),
+        (re.compile(r'{\\bf (.*?)}'), r"\1"),
+    )
```

Comment:
Suggestion: Do we know for sure that this is a complete list of equations? If not, a more general regex might be better:
```suggestion
        # Is used for matching brackets in mathjax, should not be required in plain text.
        ("\\left", ""),
        ("\\right", ""),
    )
    regex_replacements = (
        # Makes text bold, so not required in plain text.
        (re.compile(r'\\mathbf{(.*?)}'), r"\1"),
        (re.compile(r'{\\bf (.*?)}'), r"\1"),
        # Match equation-like syntax, e.g \cos(y) or \csc(3 + 3)
        (re.compile(r'\\(\w+)(\(.*?\))'), r"\1\2"),
    )
```

Reply:
@pomegranited I specifically converted `sin, cos` etc. as I know that they are displayed as is without the `\` prefix when rendered using mathjax. Having a generic regex like above would also replace parts that can be handle by `unicodeit` for example: `\sqrt` which is converted to `√`.

Reply:
No worries! I'm ok with explicitly calling out the things we change.